source/common/common/key_value_store_base.cc

```diff
@@ -0,0 +1,102 @@
+#include "source/common/common/key_value_store_base.h"
+
+namespace Envoy {
+namespace {
+
+// Removes a length prefixed token from |contents| and returns the token,
+// or returns absl::nullopt on failure.
+absl::optional<absl::string_view> getToken(absl::string_view& contents) {
+  const auto it = contents.find("\n");
+  if (it == contents.npos) {
+    ENVOY_LOG_MISC(warn, "Bad file: no newline");
+    return {};
+  }
+  uint64_t length;
+  if (!absl::SimpleAtoi(contents.substr(0, it), &length)) {
+    ENVOY_LOG_MISC(warn, "Bad file: no length");
+    return {};
+  }
+  contents.remove_prefix(it + 1);
+  if (contents.size() < length) {
+    ENVOY_LOG_MISC(warn, "Bad file: insufficient contents");
+    return {};
+  }
+  absl::string_view token = contents.substr(0, length);
+  contents.remove_prefix(length);
+  return token;
+}
+
+bool tokenizeContents(absl::string_view contents,
```

Comment:
Mind adding a comment here? It seems like what this is doing is reading a serialized store from |contents| into the map in |store| is that right? If so, I might be inclined to call this parseContents.

Reply:
Moved :-)